acceptance/tests/provider/package/gem.rb

```diff
@@ -79,6 +80,34 @@
       on(agent, "#{puppet_gem_command} uninstall #{package}")
     end
 
+    step "Install a gem package in a certain min max range" do
+      package_manifest1 = resource_manifest('package', package, { ensure: '>0.5 <0.7', provider: 'gem' } )
+      package_manifest2 = resource_manifest('package', package, { ensure: '>0.7 <0.8.1',  provider: 'gem' } )
+
+      # Ensure the package is not already installed
+      on(agent, "#{puppet_gem_command} uninstall #{package}", :accept_all_exit_codes => true)
+
+      # Make a fresh package install (with version between 0.5 and 0.7)
+      apply_manifest_on(agent, package_manifest1, :expect_changes => true) do
+        list = on(agent, "#{puppet_gem_command} list").stdout
+        match = list.match(/#{package} \((.+)\)/)
+        installed_version = match[1] if match
+        assert_match(installed_version, '0.6.0')
+      end
+
+      # Reapply same manifest and expect no changes
+      apply_manifest_on(agent, package_manifest1, :catch_changes => true)
+
+      # Install besides existing package (with version between 0.7 and 0.8.1) and expect changes
+      apply_manifest_on(agent, package_manifest2, :expect_changes => true) do
+        list = on(agent, "#{puppet_gem_command} list").stdout
+        match = list.match(/#{package} \((.+)\)/)
+        installed_versions = match[1] if match
+        assert_match(installed_versions, '0.8.0, 0.6.0')
+      end
+      on(agent, "#{puppet_gem_command} uninstall #{package} --all")
+    end
```

Comment:
the test should leave the system in the same state it was found => package should be uninstalled if it was not present at the beginning